conan/tools/files/files.py

```diff
@@ -222,3 +223,13 @@ def save_toolchain_args(content, generators_folder=None):
     toolchain_config[CONAN_TOOLCHAIN_ARGS_SECTION] = content_
     with open(args_file, "w") as f:
         toolchain_config.write(f)
+
+
+@contextmanager
+def chdir(newdir):
```

Comment:
``mkdir()`` uses ``conanfile`` as first argument, shouldn't we follow a similar pattern for ``chdir()``? Or remove the ``mkdir`` first argument.